app/models/course.rb

```diff
@@ -39,6 +39,46 @@ def pct_correct_by_user(user)
     return 100.0 * right / total
   end
 
+  def timeseries_by_user(user)
```

Comment:
I hate that these two functions are so similar "sandwich" functions since they are almost identical but one line in the middle, but I don't see an immediate remedy